### PR TITLE
✨Mysql brute force module protocol

### PR DESCRIPTION
**Pull request by Mrinank-Bhowmick**
# Motivation and Context
## Changes proposed in this pull request
* Added Mysql brute module protocol , taken reference from [ftp_brute](https://github.com/OWASP/Nettacker/blob/master/modules/brute/ftp.yaml) .
* Changed Dockerfile to add a line to install [mysql-connector-python](https://pypi.org/project/mysql-connector-python/) library.  
  -  here it is placed separately as it will show error if placed in requirements.txt

### Reviewers
@Ali-Razmjoo
@securestep9

#### Checklist
- [x] I have followed the [Contributor Guidelines](https://github.com/OWASP/Nettacker/wiki/Developers#contribution-guidelines).
- [x] The code has been thoroughly tested in my local development environment with flake8 and pylint.
- [x] The code is Python 3 compatible.
- [x] The code follows the PEP8 styling guidelines with 4 spaces indentation.
- [x] This Pull Request relates to only one issue or only one feature
- [ ] I have referenced the corresponding issue number in my commit message
- [x] I have added the relevant documentation.
- [x] My branch is up-to-date with the Upstream master branch.

#### Your development environment
- OS: `Docker: Debian GNU/Linux 11 (bullseye)`
- OS Version: `11`
- Python Version: `3.9`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,7 +5,9 @@ COPY . .
 RUN mkdir -p .data/results
 RUN apt-get update
 RUN apt-get install -y < requirements-apt-get.txt
-RUN pip3 install --upgrade pip
+RUN pip3 install --upgrade pip 
+# Below command is only used here separately as it will show error if placed in requirements.txt 
+RUN pip3 install mysql-connector-python
```
Comment (Ali-Razmjoo) [inline Dockerfile:10]: this must be in the requirements.txt

File: core/module_protocols/Mysql.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
```
Comment (Ali-Razmjoo) [inline core/module_protocols/Mysql.py:1]: this file must be renamed to core_mysql.py

File: modules/brute/Mysql.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  severity: 3
+  description: Mysql Bruteforcer
+  reference:
+  profiles:
+    - brute
+    - brute_force
+    - Mysql
+
+payloads:
+  - library: Mysql
```
Comment (Ali-Razmjoo) [inline modules/brute/Mysql.yaml:13]: after renaming the lib fix this, use lowercase letter

File: modules/brute/Mysql.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    steps:
+      - method: Mysql_brute_force
```
Comment (Ali-Razmjoo) [inline modules/brute/Mysql.yaml:15]: function name must be lower case

File: core/module_protocols/Mysql.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import copy
+import mysql.connector as sqltor
+
+# from core.utility import reverse_and_regex_condition
+from core.utility import process_conditions
+from core.utility import get_dependent_results_from_database
+from core.utility import replace_dependent_values
+
+
+# def response_conditions_matched(sub_step, response):
```
Comment (Ali-Razmjoo) [inline core/module_protocols/Mysql.py:13]: let's clean up unnecessary comments, should do in other modules too.

**Review by Ali-Razmjoo — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (Mrinank-Bhowmick) [reply Dockerfile:10]: I tried this method earlier but it didn't worked, that's why I found this solution and ran `RUN pip3 install mysql-connector-python` , as a separate line in dockerfile.

File: core/module_protocols/Mysql.py (hunk 5 above)
Comment (Mrinank-Bhowmick) [reply core/module_protocols/Mysql.py]: Sure I will do it for this file as well as for other files too.

File: modules/brute/Mysql.yaml (hunk 4 above)
Comment (Mrinank-Bhowmick) [reply modules/brute/Mysql.yaml:15]: ok

**Comment by Mrinank-Bhowmick** [comment]
@Ali-Razmjoo is it ok now?

File: core/module_protocols/core_mysql.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+import copy
+import mysql.connector as sqltor
+from core.utility import process_conditions
+from core.utility import get_dependent_results_from_database
+from core.utility import replace_dependent_values
+
+class NettackMYSQLLib:
+    def mysql_brute_force(host, ports, usernames, passwords, timeout):
+        connection = sqltor.connect(host=host,user=usernames,passwd=passwords)
```
Comment (Ali-Razmjoo) [inline core/module_protocols/core_mysql.py:12]: please apply the port here.

**Pull request closed without merging**